core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/PartitionAwarePolicy.java

```diff
@@ -30,7 +30,7 @@
 import com.datastax.oss.driver.api.core.type.SetType;
 import com.datastax.oss.driver.api.core.type.UserDefinedType;
 import com.datastax.oss.driver.internal.core.loadbalancing.DefaultLoadBalancingPolicy;
-import com.datastax.oss.driver.internal.core.util.collection.QueryPlan;
+import com.datastax.oss.driver.internal.core.util.collection.QueryPlanNew;
```

Comment:
Why did we need a new class `QueryPlanNew`, could we have used `QueryPlanBase`. This is a major change, probably we'll need to add a test case if there is no integration test coverage.

Reply:
We need to verify what each of these new QueryPlan implementations provides - QueryPlanBase, LazyQueryPlan, EmptyQueryPlan.

Reply:
QueryPlanBase is also an abstract class and hence cannot be instantiated in the PartitionAwarePolicy.java. I will look into LazyQueryPlan and EmptyQueryPlan.